src/udiskslinuxpartition.c

```diff
@@ -296,94 +288,57 @@ handle_set_flags (UDisksPartition       *partition,
                                                     invocation))
     goto out;
 
-  device_name = udisks_daemon_util_escape_and_quote (udisks_block_get_device (partition_table_block));
+  device_name = g_strdup (udisks_block_get_device (partition_table_block));
+  partition_name = g_strdup (udisks_block_get_device (block));
+
+  /* hold a file descriptor open to suppress BLKRRPART generated by the tools */
+  fd = open (udisks_block_get_device (block), O_RDONLY);
+
   if (g_strcmp0 (udisks_partition_table_get_type_ (partition_table), "gpt") == 0)
     {
-      /* TODO: implement once
-       * https://github.com/rhinstaller/libblockdev/issues/80 is resolved.
-       */
-      command_line = g_strdup_printf ("sgdisk --attributes %u:=:0x%08x%08x %s",
-                                      udisks_partition_get_number (partition),
-                                      (guint32) (flags >> 32),
-                                      (guint32) (flags & 0xffffffff),
-                                      device_name);
+      if (flags & 1) /* 1 << 0 */
+          bd_flags |= BD_PART_FLAG_GPT_SYSTEM_PART;
+      if (flags & 0x1000000000000000) /* 1 << 60 */
+          bd_flags |= BD_PART_FLAG_GPT_READ_ONLY;
+      if (flags & 0x4000000000000000) /* 1 << 62 */
+          bd_flags |= BD_PART_FLAG_GPT_HIDDEN;
+      if (flags & 0x8000000000000000) /* 1 << 63 */
```

Comment:
Why not use the bit-shifts directly instead of just in the comments for crazy numbers?